### DISCUSSION TITLE
Compile *.c and *.s files together in Eclipse C/C++

### DISCUSSION THREAD

**Discussion by sedevdkw** [eclipse-cdt/cdt#795] [Q&A] [answered]
My project contains a collection of .h, .c, and .s files. When I attempt to compile using _arm-none-eabi-gcc_, only the .c files are compiled into object files, while the compiler doesn't process the .s files. This results in errors during the linking stage, as all the .c and .s files need to be compiled before being linked.

I have successfully compiled in the command line using _arm-none-eabi-gcc_ with the following command:

`arm-none-eabi-gcc -march=armv7-r+fp -mtune=cortex-r5 -mfpu=fpv4-sp-d16 --specs=nosys.specs -T HL_sys_link.ld -Iinclude/ source/*.c source/*.s -o output.o`

However, I am unsure how to instruct the toolchain in Eclipse C/C++ to ensure that .s files are compiled as well. Could anyone please assist me with this?

Thank you in advance.

**Comment by jld01** [comment, marked as answer]
Hi @sedevdkw, the CDT managed build system will invoke the GNU assembler on `*.S` and `*.sx` files by default. If you wish to also build `*.s` files, you will need to add a new file type via the project properties dialog:
![image](https://github.com/eclipse-cdt/cdt/assets/45816183/da54b103-b04f-4169-a158-08f15bbac4c2)
Does this answer your question?

**Reply by sedevdkw** [reply]
Thanks @jld01 that resolved the issue.
